File: Types/Response.cs
namespace RainerBlog.Types;

public class BaseResponse
{
    public int Code { get; set; } = 200;
    public string Message { get; set; } = "";
    public string Result { get; set; } = "";
}

public class CommonResponse<T> : BaseResponse
{
    public T Data { get; set; }
    
    public CommonResponse(T data)
    {
        Data = data;
    }
}

File: src/types/response.ts
class BaseResponse{
    constructor(
        public code: number,
        public message: string,
        public result: string
    ){}
}

class CommonResponse extends BaseResponse{
    constructor(
        public code: number,
        public message: string,
        public result: string,
        public data: any
    ){
        super(code, message, result);
    }
}